doc/contributing/coding_style_cpp.rst

```diff
@@ -41,3 +41,68 @@ Includes
 * Other includes should use absolute paths::
 
     #include <libdnf/.../*.hpp>
+
+Line breaks
+-----------
+clang-format doesn't enforce number of line breaks between top level definitions, hence the style of these is on a best-effort basis. Please try to adhere to the following conventions:
+
+* Single line break between top-level copyright comment, include guards and includes
+* Double line break before and between namespace blocks.
+* Single line break inside a namespace block at the start and end
+* Double line break between unrelated or loosely related definitions, single or no line break between related definitions
+* Single line break between a namespace block and an include guard #endif
+* Function/method definitions should be separated by a double line break
```

Comment:
I think I've been using 2 line breaks almost everywhere at the top level.
It's easy to remember and follow, it's also quite readable.
The only exception is a namespace block with forward declarations that has no empty lines in it to keep it compact.

Reply:
My only issue with 2 breaks everywhere is that it unnecessarily elongates the files and the result is less meaningful lines fit on my screen. Hence the single lines in my version to squash everything together a bit.